File: server/Models/Album.cs
using System.ComponentModel.DataAnnotations;
namespace postitfullstack.Models;

public class Album : RepoItem<int>
{
    public string CreatorId { get; set; }
    [MinLength(3), MaxLength(25)] public string Title { get; set; }
    [MinLength(15), MaxLength(250)] public string Description { get; set; }
    [Url, MaxLength(2000)] public string CoverImg { get; set; }
    public bool IsArchived { get; set; }
    public string Category { get; set; }
    public Profile Creator { get; set; }
}

File: client/src/models/Album.js
import { Profile } from "./Account.js"

export class Album {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.title = data.title
    this.coverImg = data.coverImg
    this.archived = data.archived
    this.description = data.description
    this.category = data.category

    this.creator = new Profile(data.creator)
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.watcherCount = data.watcherCount
  }
}
